sssd_test_framework/utils/sssd.py

```diff
@@ -783,3 +784,62 @@ def autofs(self) -> None:
         """
         self.sssd.authselect.select("sssd")
         self.sssd.enable_responder("autofs")
+
+    def proxy(
+        self,
+        proxy: Literal["files", "ldap"] = "files",
+        provider: str | list[str] = "id",
+        proxy_pam_target: str = "system-auth",
+        proxy_pam_stack: str | None = None,
+        server_hostname: str | None = None,
+        domain: str | None = None,
+    ):
+        """
+        :param proxy: ``ldap`` or ``files``
+        :type proxy: str
+        :param provider: SSSD providers (id, auth, chpass, ...)
+        :type provider: str | list[str]
+        :param proxy_pam_target: proxy_pam_target for sssd
+        :type proxy_pam_target: str
+        :param proxy_pam_stack: Config for pam stack
+        :type proxy_pam_stack: str
+        :param server_hostname: LDAP (LDAP server hostname)
+        :type server_hostname: str
+        :param domain: Existing domain name, defaults to None (= default domain)
+        :type domain: str | None, optional
+        """
+        if domain is None:
+            domain = self.sssd.default_domain
+
+        if domain is None:
+            raise ValueError("No domain specified!")
+
+        if proxy == "ldap":
+            if proxy_pam_target is None:
+                proxy_pam_target = "sssdproxyldap"
+
+            if proxy_pam_target is None:
+                raise ValueError("No proxy_pam_target specified!")
+
+            if proxy_pam_stack is None:
+                raise ValueError("No proxy_pam_stack specified!")
+
+            if server_hostname is None:
+                raise ValueError("No server_hostname specified!")
+
+            self.sssd.fs.write(f"/etc/pam.d/{proxy_pam_target}", proxy_pam_stack)
```

Comment:
You want to write this for files as well `if proxy_pam_stack is not None`.

Reply:
For files its not need, it will be :  "system-auth"

Reply:
id_provider="proxy",
proxy_lib_name="files",
proxy_pam_target="system-auth"

Reply:
Yes, but what if you want to provide custom pam target and custom pam stack?